keystorage/hashicorp/src/main/java/tech/pegasys/signers/hashicorp/HashicorpConnection.java

```diff
@@ -49,61 +43,28 @@ public String fetchKey(final KeyDefinition key) {
-              request.putHeader("X-Vault-Token", key.getToken());
-              request.setChunked(false);
-              request.exceptionHandler(futureResponse::completeExceptionally);
-              request.setTimeout(requestTimeoutMs);
-              request.end();
-            })
-        .onFailure(futureResponse::completeExceptionally);
-
+  private Map<String, String> fetchKeyValuesFromVault(final KeyDefinition keyDefinition) {
+    final URI vaultReadURI =
+        connectionParameters.getVaultURI().resolve(keyDefinition.getKeyPath()).normalize();
+    HttpRequest httpRequest =
+        HttpRequest.newBuilder(vaultReadURI)
+            .header("X-Vault-Token", keyDefinition.getToken())
+            .header("Content-Type", "application/json")
+            .timeout(Duration.ofMillis(connectionParameters.getTimeoutMilliseconds()))
+            .GET()
+            .build();
+    HttpResponse<String> response;
     try {
-      return futureResponse.get();
-    } catch (final ExecutionException e) {
-      final Throwable underlyingFailure = e.getCause();
-      if (underlyingFailure instanceof HashicorpException) {
-        throw (HashicorpException) e.getCause();
-      } else if (underlyingFailure instanceof TimeoutException) {
-        throw new HashicorpException(
-            "Hashicorp Vault failed to respond within expected timeout.", underlyingFailure);
-      } else if (underlyingFailure instanceof SSLException) {
-        throw new HashicorpException("Failed during SSL negotiation.", underlyingFailure);
-      }
-      throw new HashicorpException(ERROR_HTTP_CLIENT_CALL, underlyingFailure);
-    } catch (final InterruptedException e) {
-      throw new HashicorpException("Waiting for Hashicorp response was terminated unexpectedly");
+      response = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+    } catch (IOException | InterruptedException | RuntimeException e) {
+      throw new HashicorpException("Error communicating to Hashicorp vault: " + e.getMessage(), e);
     }
-  }
-
-  private void responseHandler(
-      final CompletableFuture<Map<String, String>> future, final HttpClientResponse response) {
 
-    final int statusCode = response.statusCode();
-    if (statusCode != HttpResponseStatus.OK.code()) {
-      future.completeExceptionally(
-          new HashicorpException(String.format("Invalid Http Status code %d", statusCode)));
-    } else {
-
-      response.bodyHandler(
-          buffer -> {
-            try {
-              final Map<String, String> kvMap = HashicorpKVResponseMapper.from(buffer.toString());
-              future.complete(kvMap);
-            } catch (final Exception e) {
-              future.completeExceptionally(e);
-            }
-          });
+    if (response.statusCode() < 200 || response.statusCode() > 299) {
+      throw new HashicorpException(
+          String.format("Invalid Http Status code %d", response.statusCode()));
```

Comment:
It would be clearer to say something similar to other exception mentioning hashicorp value. Maybe "Error fetching key from Hashicorp vault due to invalid http status code"